src/ExpressionEvaluator/Core/Source/ExpressionCompiler/PseudoVariableUtilities.cs

```diff
@@ -73,11 +72,38 @@ internal static bool TryParseVariableName(string name, bool caseSensitive, out P
             }
 
             kind = PseudoVariableKind.None;
-            index = -1;
             id = null;
+            index = -1;
             return false;
         }
 
+        internal static bool TryParseReturnValueIndex(string name, out int index)
+        {
+            Debug.Assert(name.StartsWith("$ReturnValue", StringComparison.OrdinalIgnoreCase));
+            const int prefixLength = 12; // "$ReturnValue"
+            int n = name.Length;
+            index = 0;
+            if (n == prefixLength)
+            {
+                return true;
+            }
+            return (n > prefixLength) &&
+                int.TryParse(name.Substring(prefixLength), NumberStyles.None, CultureInfo.InvariantCulture, out index);
+        }
+
+        internal static DkmClrCompilationResultFlags GetLocalResultFlags(this Alias alias)
+        {
+            switch (alias.Kind)
+            {
+                case PseudoVariableKind.Exception:
+                case PseudoVariableKind.StowedException:
+                case PseudoVariableKind.ReturnValue:
+                    return DkmClrCompilationResultFlags.ReadOnlyResult;
+                default:
```

Comment:
We check for valid kind in CreatePlaceholderLocal. Since this method is for ResultFlags only, it seems less necessary here.